KarhooSDK/Network/HttpClient/TokenRefreshingHttpClient/Interactor/KarhooRefreshTokenInteractor.swift

```diff
@@ -140,10 +173,57 @@ final class KarhooRefreshTokenInteractor: RefreshTokenInteractor {
             return true
         }
         let timeToExpiration = date.timeIntervalSince1970 - Date().timeIntervalSince1970
-        return timeToExpiration < Constants.MaxTimeIntervalToRefreshToken
+        return timeToExpiration < Constants.refreshBuffer
+    }
+
+    // MARK: Timers scheduling
+
+    private func scheduleRefreshTokenTimer() {
+        guard let credentials = dataStore.getCurrentCredentials() else {
+            assertionFailure("Credentials should be set at this stage")
+            return
+        }
+        invalidateRefreshTokenTimer()
+
+        let secondsToRefresh: TimeInterval = max(0, (credentials.expiryDate?.timeIntervalSinceNow ?? 0) - Constants.refreshBuffer)
+
+        refreshTokenTimer = Timer.scheduledTimer(
+            withTimeInterval: secondsToRefresh,
+            repeats: false,
+            block: { [weak self] _ in
+                self?.refreshToken { result in
+                    print("KarhooRefreshTokenInteractor.proactivelyRefreshToken result: \(result)")
+                }
+            }
+        )
+        RunLoop.main.add(refreshTokenTimer!, forMode: .common)
     }
 
-    private struct Constants {
-        static let MaxTimeIntervalToRefreshToken = TimeInterval(30) // 30 sec
+    /// Schedule a timer that will return extenal authentition error if given time interval will be reached. This is designed as fallback in case of invalid DPs implementation of the `requireSDKAuthentication` method.
+    private func scheduleExtenalAuthInvalidationTimer() {
+        invalidateExternalAuthTimer()
+        externalAuthInvalidationTimer = Timer.scheduledTimer(
+            withTimeInterval: Constants.allowedExternalAuthTimeInterval,
+            repeats: false,
+            block: { [weak self] _ in
+                guard let self else { return }
```

Comment:
This line is failing for me with error: "Variable binding in a condition requires an initializer".
Shouldn't be:`guard let self = self else { return }` ?

Reply:
New Swift 5.7 syntax allowed to use shortage. This is why CICD build passed successfully.